torch/distributions/wishart.py

```diff
@@ -250,32 +251,44 @@ def log_prob(self, value):
         nu = self.df  # has shape (batch_shape)
         p = self._event_shape[-1]  # has singleton shape
         return (
-            - nu * p * _log_2 / 2
+            - 0.5 * nu * p * _log_2
             - nu * self._unbroadcasted_scale_tril.diagonal(dim1=-2, dim2=-1).log().sum(-1)
-            - torch.mvlgamma(nu / 2, p=p)
-            + (nu - p - 1) / 2 * torch.linalg.slogdet(value).logabsdet
-            - torch.cholesky_solve(value, self._unbroadcasted_scale_tril).diagonal(dim1=-2, dim2=-1).sum(dim=-1) / 2
+            - torch.mvlgamma(0.5 * nu, p=p)
+            + 0.5 * (nu - p - 1) * torch.linalg.slogdet(value).logabsdet
+            - 0.5 * torch.cholesky_solve(value, self._unbroadcasted_scale_tril).diagonal(dim1=-2, dim2=-1).sum(dim=-1)
         )
 
     def entropy(self):
         nu = self.df  # has shape (batch_shape)
         p = self._event_shape[-1]  # has singleton shape
         V = self.covariance_matrix  # has shape (batch_shape x event_shape)
+
         return (
             (p + 1) * self._unbroadcasted_scale_tril.diagonal(dim1=-2, dim2=-1).log().sum(-1)
-            + p * (p + 1) * _log_2 / 2
-            + torch.mvlgamma(nu / 2, p=p)
-            - (nu - p - 1) / 2 * _mvdigamma(nu / 2, p=p)
-            + nu * p / 2
+            + 0.5 * p * (p + 1) * _log_2
+            + torch.mvlgamma(0.5 * nu, p=p)
+            - 0.5 * (nu - p - 1) * torch.digamma(
+                0.5 * (
+                    nu.unsqueeze(-1)
+                    - torch.arange(p, dtype=nu.dtype, device=nu.device).expand(nu.shape + (-1,))
+                )
+            ).sum(-1)
+            + 0.5 * nu * p
         )
 
     @property
     def _natural_params(self):
+        nu = self.df  # has shape (batch_shape)
+        p = self._event_shape[-1]  # has singleton shape
+
         return (
-            0.5 * self.df,
             - 0.5 * self.precision_matrix,
+            0.5 * (nu - p - 1),
```

Comment:
This looks like a correct but unnecessary change.

Reply:
You're right. I can clean this part.